schema/v1.1.0/dataset_config_models_extended.py

```diff
@@ -0,0 +1,321 @@
+    async with aiohttp.ClientSession() as session, session.head(url) as response:
+        return doi, response.status == 200
+
+
+@alru_cache
+async def lookup_empiar(empiar_id: str) -> Tuple[str, bool]:
+    url = f"https://www.ebi.ac.uk/empiar/api/entry/{empiar_id}/"
+    async with aiohttp.ClientSession() as session, session.head(url) as response:
+        return empiar_id, response.status == 200
+
+
+@alru_cache
+async def lookup_emdb(emdb_id: str) -> Tuple[str, bool]:
+    url = f"https://www.ebi.ac.uk/emdb/api/entry/{emdb_id}"
+    async with aiohttp.ClientSession() as session, session.head(url) as response:
+        return emdb_id, response.status == 200
+
+
+PUBLICATION_REGEXES_AND_FUNCTIONS = {
+    "doi": (r"^(doi:)?10\.[0-9]{4,9}/[-._;()/:a-zA-Z0-9]+$", lookup_doi),
+    "empiar": (r"^EMPIAR-[0-9]{5}$", lookup_empiar),
+    "emdb": (r"^EMD-[0-9]{4,5}$", lookup_emdb),
+}
+
+
+async def validate_publication_lists(publication_list: List[str]) -> List[str]:
+    # list of invalid publications
+    invalid_publications: List[str] = []
+    new_publications: Dict[str, Set[str]] = {
+        publication_type: set() for publication_type in PUBLICATION_REGEXES_AND_FUNCTIONS
+    }
+
+    for publication in publication_list:
+        for publication_type, (regex, _) in PUBLICATION_REGEXES_AND_FUNCTIONS.items():
+            if re.match(regex, publication):
+                # edge case for DOI, remove the doi: prefix
+                if publication_type == "doi":
+                    publication = publication.replace("doi:", "")
+                new_publications[publication_type].add(publication)
+                break
+
+    tasks = []
+    for publication_type, (_, validate_function) in PUBLICATION_REGEXES_AND_FUNCTIONS.items():
+        if publication_type in new_publications:
+            tasks += [validate_function(entry) for entry in list(new_publications[publication_type])]
+
+    results = await asyncio.gather(*tasks)
+    invalid_publications += [publication for publication, valid in results if not valid]
+
+    return invalid_publications
+
+
+# ==============================================================================
+# Annotation Author Validation
+# ==============================================================================
+
+
+def validate_authors_status(authors: List[Author]) -> List[ValueError]:
+    errors = []
+    primary_author_status_count = 0
+    corresponding_author_status_count = 0
+    for author in authors:
+        if author.primary_author_status:
+            primary_author_status_count += 1
+        if author.corresponding_author_status:
+            corresponding_author_status_count += 1
+
+    if primary_author_status_count == 0:
+        errors.append(ValueError("Annotation must have at least one primary author"))
+    if corresponding_author_status_count == 0:
+        errors.append(ValueError("Annotation must have at least one corresponding author"))
+
```

Comment:
doing this so that both ValueErrors can be returned if both fail (so one ValueError doesn't short-circuit the function and prevent the other one from getting checked)